### ISSUE TITLE
[eatshalal] wrong language for ingredient list

### ISSUE THREAD

**Issue by raphael0202** [openfoodfacts/contributor-quality-issues#1]
[eatshalal](https://world.openfoodfacts.org/editor/eatshalal) app replaces a valid French ingredient list by an english one, breaking ingredient analysis and NOVA computation.
This affects very popular products, such as: https://world-fr.openfoodfacts.org/produit/3017620422003/nutella-ferrero?rev=544

**Comment by raphael0202** [comment]
Response received today, saying this is going to be fixed ASAP.

**Comment by MohamedFBoussaid** [comment]
I am seeing Smooth App is using this : 
`OpenFoodFactsLanguage getLanguage() => LanguageHelper.fromJson(languageCode);`
I will try this one.

**Comment by raphael0202** [comment]
I'm not really familiar with the flutter SDK, maybe @g123k can help you on this.
But on the API, the `lang` field is the product main language (see https://openfoodfacts.github.io/openfoodfacts-server/api/ref-v2/#cmp--schemas-product-base).

**Comment by raphael0202** [comment]
Another thing, do you also own the app described in this issue? https://github.com/openfoodfacts/contributor-quality-issues/issues/2
It's also a halal app with a language field issue.

**Comment by MohamedFBoussaid** [comment]
@raphael0202, no I have only this app (Halal & Healthy) that is updating OFF DB 

I guess I have found the issue, I was setting the lang for the product, but not on the `saveProduct()` call. I have updated the code, but it is still not working, I guess I am missing something. Maybe @g123k can help me ? 

In meantime, and until I fix this, I have disabled the possibility for users to update products, I will release it now.

**Comment by MohamedFBoussaid** [comment]
@g123k , @raphael0202 , I have found the issue and fix it. 
I will `saveProduct` using `productNameInLanguages` and `ingredientsTextInLanguages` instead of `productName` and `ingredientsTextInLanguages`

I have pushed a release, and this should be fixed. 

I have test it with : 
https://fr.openfoodfacts.org/produit/3017620422003/nutella-ferrero. (updating the product for EN and AR lang) 
https://sa.openfoodfacts.org/mountaj/8595229909286/granola-honey-nuts (updating the product for ES and AR lang) 
https://sa.openfoodfacts.org/mountaj/20724696/californian-almond-alesto (updated product for AR, without impacting other lang) 
https://sa.openfoodfacts.org/mountaj/8001505005592/nocciolata-p%C3%A2te-%C3%A0-tartiner-au-cacao-et-noisettes-rigoni-di-asiago  (updated product for AR, without impacting other lang)
